src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let men = people.filter(person => person.sex === 'm');
+
+  if (arguments.length > 1) {
+    men = men.filter(man => century === Math.ceil(man.died / 100));
+  }
+
+  men = men.map(man => man.died - man.born);
```

Comment:
creete another variable for those actions, men is for other data

Reply:
agree 100%